config/initializers/user_class_extensions.rb

```diff
@@ -3,9 +3,12 @@
     Spree.user_class.class_eval do
 
       # Override of spree method to ignore orders associated with account_invoices
-      def last_incomplete_spree_order
-        spree_orders.incomplete.where("id NOT IN (?)", account_invoices.map(&:order_id)).order('created_at DESC').first
+      def last_incomplete_spree_order_with_ignoring_account_invoices
+        account_invoice_order_ids = account_invoices.map(&:order_id)
```

Comment:
Using `#map()` you lose the ActiveRecord magic generating a longer SQL query than needed. Compare the current query with the one generated by `account_invoices.pluck(:order_id)` [doc](https://apidock.com/rails/v3.2.1/ActiveRecord/Calculations/pluck).